src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -212,15 +224,22 @@
   (async/go (try
               (loop []
                 (let [task (async/<! lifecycle-chan)]
-                  (if (= task :shutdown)
-                    (do
-                      (log/debug "Received shutdown command, lifecycle worker exiting.")
-                      :done)
-                    (do
-                      (log/debug "Lifecycle worker executing lifecycle task.")
-                      (task)
-                      (log/debug "Lifecycle worker completed lifecycle task; awaiting next task.")
-                      (recur)))))
+                  (schema/validate LifeCycleTask task)
+                  (let [{:keys [type task-function]} task]
+                    (condp #(contains? %1 %2) type
+                      #{:shutdown}
+                      (do
+                        (log/debug "Received shutdown command, lifecycle worker exiting.")
+                        :done)
+
+                      #{:boot :restart}
+                      (do
+                        (log/debugf "Lifecycle worker executing %s lifecycle task." type)
+                        (task-function)
+                        (log/debugf "Lifecycle worker completed %s lifecycle task; awaiting next task." type)
+                        (recur))
+
+                      (throw (IllegalStateException. "Unrecognized lifecycle task:" task))))))
```

Comment:
It doesn't seem like it will be possible to get here, since the schema/validate on line 227 (https://github.com/puppetlabs/trapperkeeper/pull/223/commits/59347aa16f46a61fa77a41d4518e9fe728543f29#diff-452afe9e02bf063094d280de34cf34bcR227) will validate that `:type` is one of `:shutdown`, `:boot`, and `:restart`.